pkg/kubelet/dockershim/libdocker/fake_client.go

```diff
@@ -145,6 +149,24 @@ func (f *FakeDockerClient) appendContainerTrace(traceCategory string, containerN
 	}
 }
 
+// WriteDockerConfigFile creates a config file that NewDockerKeyring will pick up. It returns
+// a function that will delete the file, so the caller can `defer` it.
+func (f *FakeDockerClient) WriteDockerConfigJSONFile(cfg string) (error, func()) {
```

Comment:
Reminder: make sure you pass absolute paths to the function so that ReadDocker*File functions don't mess with them.